common/proto-conv/src/share_from_to_protobuf_impl.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This mod is the key point about compatibility.
+//! Everytime update anything in this file, update the `VER` and let the tests pass.
+
+use std::collections::BTreeMap;
+
+use common_datavalues::chrono::DateTime;
+use common_datavalues::chrono::Utc;
+use common_meta_app::schema;
+use common_meta_app::share as mt;
+use common_protos::pb;
+use enumflags2::BitFlags;
+
+use crate::check_ver;
+use crate::FromToProto;
+use crate::Incompatible;
+use crate::MIN_COMPATIBLE_VER;
+use crate::VER;
+
+impl FromToProto<pb::ShareNameIdent> for mt::ShareNameIdent {
```

Comment:
There should be a test to ensure compatibility for this version, just like:
https://github.com/datafuselabs/databend/blob/658f1957256cbb8e0459ef175f5f24a54785830e/common/proto-conv/tests/it/proto_conv.rs#L199-L235

BTW, since new types are added, `VER` should be increased.

Reply:
> There should be a test to ensure compatibility for this version, just like:
> 
> https://github.com/datafuselabs/databend/blob/658f1957256cbb8e0459ef175f5f24a54785830e/common/proto-conv/tests/it/proto_conv.rs#L199-L235
> 
> BTW, since new types are added, `VER` should be increased.

@Xuanwo